core/vm/contracts.go

```diff
@@ -668,3 +670,37 @@ func (c *ecrecoverPublicKey) Run(input []byte) ([]byte, error) {
 
 	return pubKey, nil
 }
+
+// ed25519Verify implements a native Ed25519 signature verification.
+type ed25519Verify struct{}
+
+// RequiredGas returns the gas required to execute the pre-compiled contract.
+func (c *ed25519Verify) RequiredGas(input []byte) uint64 {
```

Comment:
Are these codes directly copied from celo? Any modifications that's worth mentioning?

Reply:
I think there are no major changes required. so, it should be good.